tests/src/test/scala/cats/collections/BitSetTest.scala

```diff
@@ -176,4 +177,96 @@ object BitSetTest extends Properties("BitSet") {
       val rhs = x.reverseIterator.toList
       (lhs == rhs) :| s"$lhs == $rhs"
     }
+
+  property("((x ^ y) ^ z) = (x ^ (y ^ z))") =
+    forAll { (x: BitSet, y: BitSet, z: BitSet) =>
+      val lhs = ((x ^ y) ^ z)
+      val rhs = (x ^ (y ^ z))
+      (lhs == rhs) :| s"$lhs == $rhs"
+    }
+
+  property("(x ^ y) = (y ^ x)") =
+    forAll { (x: BitSet, y: BitSet) =>
+      val lhs = (x ^ y)
+      val rhs = (y ^ x)
+      (lhs == rhs) :| s"$lhs == $rhs"
+    }
+
+  property("(x ^ x) = 0") =
+    forAll { (x: BitSet) =>
+      val lhs = (x ^ x)
+      val rhs = BitSet.empty
+      (lhs == rhs) :| s"$lhs == $rhs"
+    }
```

Comment:
can we add a law like:
```scala
(x ^ y)(z) == (x(z) ^ y(z))
```

also, should we just use the law testing in cats-algebra vs re-inlining? See https://github.com/typelevel/cats-collections/pull/172 -- it's pretty easy, but I think we need a new test class to use the laws.

Reply:
I support using the law-testing, but I'd rather do that in a follow-up if that's OK. I'll add this law.